tests/fixtures/project_with_invalid_dev_deps/pyproject.toml

```diff
@@ -0,0 +1,13 @@
+[tool.poetry]
```

Comment:
I would move this to the builders fixtures as this needs to be tested with api not with factory. Maybe name it `with_invalid_path_dep_dev` and add package to the source too so we can evaluate the built wheel.

Additionally, I would also recommend that we add a fixture for `with_invalid_path_dep_main` that can be used to ensure an error is raised if encountered.

Reply:
Do you believe strongly that I should ONLY test in test_api and not in test_factory at all? I'll remove those tests if you don't want them, but I think they're doing something slightly different, something a little less e2e and applicative but more focused on the module itself.

Reply:
I believe you are testing the wrong things at the wrong level. The change is to ensure that when called via the PEP 517 api we ignore the dev dependnecies. The change in the factory code is to enable/disable inclusion of development dependencies and hence that is what should be tested, not if the "invalid" dependencies are not raising exceptions. This is what I have indicated in my other comment.

The suggestion to move the fixture is more for consistency as that is where the builder related fixtures are at. For verifying the flag, you can simply use an existing fixture here with both main and dev dependencies and parameterised test validating both flag states.

Reply:
@abn I wrote new tests in test_api that I believe does what you are asking for. I also updated the test_factory tests which I believe also does what you are asking for. Did you get a chance to look at the new tests? Is your opinion still unchanged?

Reply:
Looks about right at first glance will review tomorrow. Thanks for the changes.